src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -872,7 +879,55 @@ int dhcp_server_handle_message(sd_dhcp_server *server, DHCPMessage *message,
 
                 /* verify that the requested address is from the pool, and either
                    owned by the current client or free */
-                if (pool_offset >= 0 &&
+                if(static_lease) {
+                        DHCPLease *lease;
+                        usec_t time_now = 0;
+                        lease = new0(DHCPLease, 1);
+                        if (!lease)
+                          return -ENOMEM;
```

Comment:
indentation off. at the most basic, please follow the coding style of the rest of the file